### DISCUSSION TITLE
Bug: macOS: flicker when restoring window with multiple tabs and multiple splits per tab

### DISCUSSION THREAD

**Discussion by kdrag0n** [ghostty-org/ghostty#4181] [Feature Requests, Ideas]
Steps to reproduce:
- Open a window with 2 tabs
- Create 2 splits in each tab
- Quit and reopen Ghostty

https://github.com/user-attachments/assets/40cf0620-3082-4ee5-aca1-55c9c8f1919a

If you frame-step the video, you can see that on relaunch, the restored window appears briefly, disappears, and then reappears. 

I think this is a regression between 599934d5 (good) and 2fb92dd4 (bad). (sorry, forgot to report earlier when I first noticed it after an update)

Config: none

**Comment by mitchellh** [comment]
Same issue as your other one.

**Reply by kdrag0n** [reply]
I don't think this is the same: the other issue is about the terminal view briefly having the wrong size, but in this case the entire macOS app window disappears momentarily, including borders / shadow / controls. It's not only the terminal surface that flickers.

**Reply by mitchellh** [reply]
Ah yes. As far as I know this is caused by us using SwiftUI. I’m not sure how we can smoothen this out but I’d love to

**Reply by npupko** [reply]
There's no such problems even with iTerm, although it's using Swift as well. Could we hope it will be fixed some day? Ghostty has a good potential, but this flickering is really disturbing

**Reply by mitchellh** [reply]
Swift != SwiftUI. I don't think iTerm is using much SwiftUI.

**Comment by maruks** [comment]
please fix this bug

**Comment by mitchellh** [comment]
I can still reproduce this on tip today, though it's only on restore. In your video it also happens live while splitting and we fixed that at some point. I'm not sure the best way to fix this, except maybe hiding the window we restore until all splits are open and initialized...
